ACPs/103-dynamic-fees/README.md

```diff
@@ -0,0 +1,133 @@
+| - | - |
+| $M$ - minimum gas price | TODO |
+| $T$ - target gas per second | TODO |
+| $L$ - max gas per second | TODO |
+| $K$ - change constant | TODO |
+
+As the network gains capacity to handle additional load (through the activation of future protocol enhancements), this algorithm can be tuned to increase the number of transactions that can be processed at a specific fee rate.
+
+#### A note on $e^x$
+
+There is a subtle reason why an exponential adjustment function was chosen: The adjustment function should be _equally_ reactive irrespective of the actual fee.
+
+Define $b_n$ as the current block's required fee, $b_{n+1}$ as the next block's required fee, and $x$ as the excess gas usage.
+
+Let's use a linear adjustment function:
+
+$$b_{n+1} = b_n + 10x$$
+
+Assume $b_n = 100$ and the current block is 1 unit above target utilization, or $x = 1$. Then, $b_{n+1} = 100 + 10 \cdot 1 = 110$, an increase of `10%`. If instead $b_n = 10,000$, $b_{n+1} = 10,000 + 10 \cdot 1 = 10,010$, an increase of `0.00001%`. The fee is _less_ reactive as the fee increases. This is because the rate of change is constant: $\frac{d}{dx}10x = 10$.
+
+Now, let's use an exponential adjustment function:
+
+$$b_{n+1} = b_n \cdot e^x$$
+
+Assume $b_n = 100$ and the current block is 1 unit above target utilization, or $x = 1$. Then, $b_{n+1} = 100 \cdot e^1 \approx 271.828$, an increase of `271%`. If instead $b_n = 10,000$, $b_{n+1} = 10,000 \cdot e^1 \approx 27,182.8$, an increase of `271%` again. The fee is _equally_ reactive as the fee increases. This is because the rate of change scales with $x$: $\frac{d}{dx}e^x = e^x$.
+
+### Block Building Procedure
+
+When a transaction is constructed on the X-Chain and P-Chain, the amount of $AVAX burned in a transaction is given by `sum($AVAX outputs) - sum($AVAX inputs)`. The amount of gas used by the transaction can also be deterministically calculated after construction. Dividing the amount of $AVAX burned by the amount of gas yields the maximum gas price that the transaction can pay.
+
+Instead of using a FIFO queue for the mempool (like the X-Chain and P-Chain do now), the mempool should be a priority queue ordered by the maximum gas price of each transaction. This ensures that higher paying transactions are included first.
+
+## Backwards Compatibility
+
+Modfication of a fee mechanism is an execution change and requires a mandatory upgrade for activation. Implementers must take care to not alter the execution behavior prior to activation.
+
+After this ACP is activated, any transaction issued on the P-Chain or X-Chain must account for the fee mechanism defined above. There is a concern of underpriced transactions filling the mempool and causing memory issues on nodes. To alleviate memory pressure, the mempool should be periodically cleared of underpriced transactions. Users are responsible for re-broadcasting underpriced transactions when the fee lowers or re-constructing their transaction to include a larger fee.
```

Comment:
I'm not sure I understand this concern, the mempools should have a maximum size and the a tx paying a higher fee should be able to evict a transaction paying a lower fee.

Reply:
adjusted the phrasing